Add idoffset argument for clustering of non-overlapped paired-end reads

astanabe opened the pull request:
I added a "idoffset" argument.
The definition of percent identity was modified to the following.

> 100 * (the number of matches - idoffset) / (the alignment length - idoffset)

This argument should be useful for clustering of non-overlapped paired-end reads as explained below.

Assuming you constructed concatenated sequences like the following,

> <- reverse_read - padding - forward_read ->

the clustered results will be OVERclustered because of padding sequence.
In such cases, giving the length of padding sequence as idoffset will produce correctly clustered results.
After the clustering, the representative or consensus sequences can be divided into forward and reverse parts using padding seqeunce.

torognes commented:
Thanks for the contribution. Could you please add some more explanation for the need for this feature? The padding sequence, is that a specific (unique) sequence that is used to pad all pairs. Could you point to more information about this approach?

astanabe commented:
I have some non-overlapped paired-end amplicon sequences and I need to pick OTUs and to assign taxa to the OTUs.
I developed analysis pipeline like the following for this data.

1. construct concatenated sequence like `<- reverse_read - padding - forward_read ->`
2. cluster by VSEARCH
3. de novo chimera removal by VSEARCH
4. split representatives of concatenated sequences based on common padding sequence
5. assign taxa to forward sequences of the representatives
6. assign taxa to reverse sequences of the representatives
7. compare and merge assignments of forward and reverse sequences

In this pipeline, common padding sequence is required for splitting.
However, VSEARCH always OVERcluster the input sequences because of padding sequence.
Thus, I added the idoffset option for VSEARCH.

Of course, this modification is very influential and you may not accept this request without careful consideration.
